source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -190,6 +253,7 @@ TEST(NiFakeServiceTests, NiFakeService_InitWithOptionsFails_NoSessionIsStored)
   EXPECT_EQ(message, status.error_message());
   nidevice_grpc::Session session = response.vi();
   EXPECT_EQ(0, session_repository.access_session(session.id(), ""));
+  EXPECT_FALSE(response.new_session_initialized());
```

Comment:
I don't think this is a valid test when the gRPC status is other than OK.

Reply:
I recommend adding an assertion that the field is true in the test method below `NiFakeService_InitWithOptionsAndResetServer_SessionIsClosed` which does initialize a new session. Or possibly choose a different one.

Reply:
Ah, good catch. Yeah, that was a bogus assertion just testing the proto message field's default value essentially.

I updated to remove this assert. I also added some more test variations to cover the success cases for the new options.

So now I assert this output is True in
`NiFakeService_InitWithOptionsSucceeds_CreatesAndStoresSession` and `NiFakeService_InitWithOptionsWithInitializeBehavior_CreatesAndStoresSession`
and assert it's False in `NiFakeServiceWithSession_InitWithOptionsWithAttachBehavior_ReturnsAlreadyInitializedSession`